packages/ubuntu_wsl_setup/test/installation_slides_page_test.dart

```diff
@@ -0,0 +1,214 @@
+        ),
+      }),
+    );
+  }
+
+  testWidgets('skip slides if server is already up', (tester) async {
+    final model = MockInstallationSlidesModel();
+    when(model.isServerUp).thenAnswer((_) => true);
+    await tester
+        .pumpWidget(buildApp((_) => buildPage(model, [const Text(title)])));
+    expect(find.byType(SlideShow), findsNothing);
+  });
+
+  testWidgets('display slides while server is not up', (tester) async {
+    MockInstallationSlidesModel model = createModel(isRegistering: true);
+    await tester
+        .pumpWidget(buildApp((_) => buildPage(model, [const Text(title)])));
+    expect(find.text(title), findsWidgets);
+    expect(find.byType(CircularProgressIndicator), findsOneWidget);
+    expect(find.text(tester.lang.unpacking), findsOneWidget);
+  });
+
+  testWidgets('display error status', (tester) async {
+    tester.binding.window.devicePixelRatioTestValue = 1.0;
+    MockInstallationSlidesModel model = createModel(hasError: true);
+    await tester
+        .pumpWidget(buildApp((_) => buildPage(model, [const Text(title)])));
+    expect(find.text(title), findsWidgets);
+    expect(find.byType(CircularProgressIndicator), findsNothing);
+    expect(find.text(tester.lang.errorTitle), findsOneWidget);
+    expect(find.text(tester.lang.errorMsg), findsOneWidget);
+  });
+
+  testWidgets('display log', (tester) async {
+    const log = 'A log';
+    MockInstallationSlidesModel model =
+        createModel(isLogVisible: true, journalSource: [log]);
+    await tester
+        .pumpWidget(buildApp((_) => buildPage(model, [const Text(title)])));
+    expect(find.text(title), findsWidgets);
+    expect(find.text(log), findsWidgets);
+  });
+
+  testWidgets('toggle log visibility up', (tester) async {
+    final model = createModel(isLogVisible: false);
+    await tester
+        .pumpWidget(buildApp((_) => buildPage(model, [const Text(title)])));
+    await tester.pumpAndSettle();
+
+    await tester.tap(find.byIcon(Icons.terminal));
+    verify(model.toggleLogVisibility()).called(1);
+  });
+
+  testWidgets('toggle log visibility down', (tester) async {
+    final model = createModel(isLogVisible: true);
+    await tester
+        .pumpWidget(buildApp((_) => buildPage(model, [const Text(title)])));
+    await tester.pumpAndSettle();
+
+    await tester.tap(find.byIcon(Icons.terminal));
+    verify(model.toggleLogVisibility()).called(1);
+  });
+
+  testWidgets('creates a model', (tester) async {
+    tester.binding.window.devicePixelRatioTestValue = 1.0;
+    final journal = MockJournalService();
+    when(journal.stream).thenAnswer((realInvocation) => Stream<String>.empty());
+    final monitor = MockSubiquityStatusMonitor();
+    when(monitor.status).thenAnswer((realInvocation) => null);
+    when(monitor.onStatusChanged).thenAnswer(
+      (realInvocation) => Stream.empty(),
+    );
+    registerService<JournalService>(() => journal);
+    registerService<SubiquityStatusMonitor>(() => monitor);
```

Comment:
`registerService` for factories that create new instances. use `registerMockService` for existing instances passed from the outside:
```dart
registerMockService<JournalService>(journal);
registerMockService<SubiquityStatusMonitor>(monitor);
```